src/main/java/org/broadinstitute/hellbender/tools/walkers/filters/VariantFiltration.java

```diff
@@ -323,9 +331,23 @@ private void filter(final VariantContext vc, final FeatureContext featureContext
         writer.add(builder.make());
     }
 
-    private GenotypesContext makeGenotypes(final VariantContext vc) {
+    private GenotypesContext makeGenotypes(final VariantContext vc, final VariantContextBuilder builder ) {
         final GenotypesContext genotypes = GenotypesContext.create(vc.getGenotypes().size());
 
+        //
+        // recompute AC, AN and AF if filtered genotypes are set to no-call
+        //
+        // occurrences of alternate alleles over all genotypes
```

Comment:
The new logic here and the logic above are substantially similar.  Is it possible to factor this so that the chromosomeCounts logic is separated from the filtering and applied afterwards.  T
hen you might be able to have a single function that does the update in both VariantFiltration and select variants.